common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -615,6 +625,162 @@ public boolean verify(@NonNull final SigningAlgorithm alg,
+            cipherOutputStream.write(plaintext.getBytes(ENCODING_UTF8));
+            cipherOutputStream.close();
+
+            // Flatten our OutputStream to an array
+            byte[] encryptedBase64Data = outputStream.toByteArray();
+
+            // Base64 encode to stringify
+            return Base64.encodeToString(encryptedBase64Data, Base64.DEFAULT);
+        } catch (final InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        } catch (final UnrecoverableEntryException e) {
+            errCode = INVALID_PROTECTION_PARAMS;
+            exception = e;
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (final KeyStoreException e) {
+            errCode = KEYSTORE_NOT_INITIALIZED;
+            exception = e;
+        } catch (final NoSuchPaddingException e) {
+            errCode = NO_SUCH_PADDING;
+            exception = e;
+        } catch (final UnsupportedEncodingException e) {
+            errCode = UNSUPPORTED_ENCODING;
+            exception = e;
+        } catch (final IOException e) {
+            errCode = IO_ERROR;
+            exception = e;
+        }
+
+        final ClientException clientException = new ClientException(
+                errCode,
+                exception.getMessage(),
+                exception
+        );
+
+        Logger.error(
+                TAG + methodName,
+                errCode,
+                exception
+        );
+
+        throw clientException;
+    }
+
+    @Override
+    public String decrypt(@NonNull final Cipher cipher,
+                          @NonNull final String ciphertext) throws ClientException {
+        final String methodName = ":decrypt";
+        final String errCode;
+        final Exception exception;
+
+        try {
+            // Load our key material
+            final KeyStore.PrivateKeyEntry privateKeyEntry = (KeyStore.PrivateKeyEntry)
+                    mKeyStore.getEntry(mKeyAlias, null);
+
+            // Get a reference to our private key (will not be loaded into app process)
+            final PrivateKey privateKey = privateKeyEntry.getPrivateKey();
+
+            // Init our cipher instance, don't use a named provider as there seems to be a mix of
+            // BoringSSL & AndroidOpenSSL
+            // https://issuetracker.google.com/issues/37091211
+            final javax.crypto.Cipher outputCipher = javax.crypto.Cipher.getInstance(cipher.toString());
+            outputCipher.init(javax.crypto.Cipher.DECRYPT_MODE, privateKey);
+
+            // Put our ciphertext into an InputStream
+            final CipherInputStream cipherInputStream = new CipherInputStream(
+                    new ByteArrayInputStream(
+                            Base64.decode(
+                                    ciphertext,
+                                    Base64.DEFAULT
+                            )
```

Comment:
Do we not have a Base64Stream running around somewhere?

Reply:
84e685554367f6eaaa4d4afccb70e478a944bc0e